src/sudoku.py

```diff
@@ -194,15 +205,26 @@ def build_grid() -> Grid:
         return sudokus[9].get_grid()
 
     @classmethod
-    def get_task(cls, speculation_depth):
-        task = cls.build_grid()
-        for index in random.sample(range(81), 81):
-            i: int = index // 9
-            j: int = index % 9
-            true_cell_value = task[i][j]
-            task[i][j] = None
+    def get_task(cls, speculation_depth, min_difficulty=0, max_difficulty=81):
+        while True:
+            task = cls.build_grid()
+            for index in random.sample(range(81), 81):
+                i: int = index // 9
+                j: int = index % 9
+                true_cell_value = task[i][j]
+                task[i][j] = None
+                sudoku = cls(task, speculation_depth)
+                sudoku.solve()
+                if not sudoku.is_solved or (
+                    sudoku._counter[sudoku._leave_equal_alternatives].solved
+                   + sudoku._counter[sudoku._exclude_equal_alternatives].solved
+                   ) > max_difficulty:
+                    task[i][j] = true_cell_value
             sudoku = cls(task, speculation_depth)
             sudoku.solve()
-            if not sudoku.is_solved:
-                task[i][j] = true_cell_value
+            if (
+                sudoku._counter[sudoku._leave_equal_alternatives].solved
+               + sudoku._counter[sudoku._exclude_equal_alternatives].solved
+               ) >= min_difficulty:
```

Comment:
Форматирование не хорошее.
Ещё разок возможные варианты:
```
foo = (foo bar
       + foo bar)
```
или
```
foo = (
    foo bar
    + foo bar
)
```


```suggestion
            if (
                sudoku._counter[sudoku._leave_equal_alternatives].solved
                + sudoku._counter[sudoku._exclude_equal_alternatives].solved
            ) >= min_difficulty:
```